Lib/http/server.py

```diff
@@ -686,10 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                ims = email.utils.parsedate_to_datetime(
```

Comment:
`email.utils.parsedate_to_datetime()` can raise an exception. It might be better just to ignore the cache rather than return an error.

Needed a cache for ill-formed "If-Modified-Since" value.